python/pyspark/sql/context.py

```diff
@@ -203,18 +203,46 @@ def registerFunction(self, name, f, returnType=StringType()):
         >>> _ = sqlContext.udf.register("stringLengthInt", lambda x: len(x), IntegerType())
         >>> sqlContext.sql("SELECT stringLengthInt('test')").collect()
         [Row(stringLengthInt(test)=4)]
+        """
+        return self.sparkSession.catalog.registerFunction(name, f, returnType)
+
+    @ignore_unicode_prefix
+    @since(2.3)
+    def registerUDF(self, name, f):
```

Comment:
I prefer not to duplicate the doc string. Maybe we can put the docstring in the user facing API (I think it's the SQLContext one?) And reference the doc string in the other one.

Or, maybe we can do sth like this if we want both docstrings

```
@ignore_unicode_prefix
@since(2.3)
def registerUDF(self, name, f):
       return self.sparkSession.catalog.registerUDF(name, f)
registerUDF.__doc__ = pyspark.sql.catalog.Catalog.registerUDF.__doc__
```

Reply:
I think that's ideally right. But let's keep it consistent with the convention used here. Let's discuss and try it separately and do it in batch.

Reply:
Ok. Sounds good to me.

Reply:
And also, seems we need to fix the examples too in this case .. `spark` -> `sqlContext`.

Reply:
The test cases are actually different. In this file, all the examples are using sqlContext.

Reply:
I am not sure about the difference between:
`spark.udf.registerUDF` `sqlContext.udf.registerUDF` and `sqlContext.registerUDF`

Seems too many ways to do the same thing...But if we indeed need to keep multiple methods, I would lean towards having comprehensive doc in one of them and have the doc for the rest to be something like 
"""
Same as :meth:...
"""

Reply:
We are deprecating sqlContext since Spark 2.0. The suggested calls should be `spark.udf.registerUDF` or `spark.catalog.registerUDF`. 

We can have a separate PR for cleaning all these method descriptions in sqlContext

Reply:
Ok. Sounds good.